src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Diagnostics/RazorTranslateDiagnosticsEndpointTest.cs

```diff
@@ -940,8 +940,8 @@ public async Task Handle_ProcessDiagnostics_Html_WithCSharpInTagHelperAttribute_
             Kind = RazorLanguageKind.Html,
             Diagnostics = new[]
             {
-                new VSDiagnostic() { Range = new Range { Start = new Position(0, addTagHelper.Length + 20),End =  new Position(0, addTagHelper.Length + 25)} },
-                new VSDiagnostic() { Range = new Range { Start = new Position(0, addTagHelper.Length + 38),End =  new Position(0, addTagHelper.Length + 47)} }
+                new VSDiagnostic() { Range = new Range { Start = new Position(1, 20),End =  new Position(1, 25)} },
+                new VSDiagnostic() { Range = new Range { Start = new Position(1, 38),End =  new Position(1, 47)} }
```

Comment:
This was passing only because we weren't checking character position within the line, and the passed in "character" did actually match up with the source position. It was wrong though and failed after I added line position check. Here we have two lines in the document, with first one being the "addTagHelpers" line of the length "addTagHelpers.Length" (happens to be 31), and the second line actully containing the text with mappings. Basically the document text looks like this
```
@addTagHelper *, TestAssembly
<button onactivate="Send" disabled="@(Something)">Hi</button>
```

Reply:
Definitely a lazy test taking advantage of the fact that we didn't validate. In an ideal world we'd get rid of all of these hard-coded numbers in tests, which are completely inscruitable, and make everything use the `TestFileMarkupParser`, but doesn't have to be now (and if you haven't worked on Roslyn you also might not be familiar with the syntax 😁)

Reply:
Didn't do it in this PR but I've seen this in other tests and will do a quick clean-up PR for this test when I have a chance. I don't like the idea of magic numbers - hard to understand what the test does.